crates/sui-core/src/traffic_controller.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::SocketAddr;
+use std::{collections::HashMap, sync::Arc};
+
+use mysten_metrics::spawn_monitored_task;
+use parking_lot::RwLock;
+use std::time::{Duration, SystemTime};
+use sui_types::traffic_control::{
+    Policy, PolicyConfig, PolicyResponse, TrafficControlPolicy, TrafficTally,
+};
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TrySendError;
+use tracing::warn;
+
+type BlocklistT = Arc<RwLock<HashMap<SocketAddr, SystemTime>>>;
+
+#[derive(Clone)]
+struct Blocklists {
+    connection_ips: BlocklistT,
+    proxy_ips: BlocklistT,
+}
+
+#[derive(Clone)]
+pub struct TrafficController {
+    tally_channel: mpsc::Sender<TrafficTally>,
+    blocklists: Blocklists,
+    //metrics: TrafficControllerMetrics, // TODO
+}
+
+impl TrafficController {
+    pub async fn spawn(policy_config: PolicyConfig) -> Self {
+        let (tx, rx) = mpsc::channel(policy_config.channel_capacity);
+        let ret = Self {
+            tally_channel: tx,
+            blocklists: Blocklists {
+                connection_ips: Arc::new(RwLock::new(HashMap::new())),
+                proxy_ips: Arc::new(RwLock::new(HashMap::new())),
+            },
+        };
+        let blocklists = ret.blocklists.clone();
+        spawn_monitored_task!(run_tally_loop(rx, policy_config, blocklists));
+        ret
+    }
+
+    pub fn tally(&self, tally: TrafficTally) {
+        // Use try_send rather than send mainly to avoid creating backpressure
+        // on the caller if the channel is full, which may slow down the critical
+        // path. Dropping the tally on the floor should be ok, as in this case
+        // we are effectively sampling traffic, which we would need to do anyway
+        // if we are overloaded
+        match self.tally_channel.try_send(tally) {
+            Err(TrySendError::Full(_)) => {
+                warn!("TrafficController tally channel full, dropping tally");
+                // TODO: metric
```

Comment:
once we have a metric on this, and can verify we aren't hitting this under normal conditions, it may be better to reject the request if the channel is full, since we are clearly overloaded in that situation.